org.openjdk.jmc.console.ext.agent/src/main/java/org/openjdk/jmc/console/ext/agent/manager/wizards/PresetManagerPage.java

```diff
@@ -58,29 +82,177 @@ public PresetManagerPage(PresetRepository repository) {
 
 		ScrolledComposite sc = new ScrolledComposite(parent, SWT.H_SCROLL | SWT.V_SCROLL);
 		Composite container = new Composite(sc, SWT.NONE);
 		sc.setContent(container);
+		container.setLayout(new GridLayout(2, false));
 
-		// TODO: create manager control here
-		container.setLayout(new FillLayout());
-		new Label(container, SWT.NONE).setText("TODO: create manager control here");
+		createPresetTable(container);
+		createButtons(container);
 
 		sc.setExpandHorizontal(true);
 		sc.setExpandVertical(true);
 		sc.setMinSize(container.computeSize(SWT.DEFAULT, SWT.DEFAULT));
 		setControl(sc);
+
+		bindListeners();
+	}
+
+	private void createPresetTable(Composite parent) {
+		tableViewer = new TableViewer(parent, SWT.V_SCROLL | SWT.BORDER | SWT.MULTI);
+		tableViewer.getControl().setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+		tableViewer.setContentProvider(new PresetTableContentProvider());
+		tableViewer.setLabelProvider(new PresetTableLabelProvider());
+		tableViewer.setInput(repository);
+	}
+
+	private void createButtons(Composite parent) {
+		Composite container = new Composite(parent, SWT.NONE);
+		container.setLayoutData(new GridData(SWT.FILL, SWT.BEGINNING, false, true));
+		GridLayout layout = new GridLayout(1, true);
+		layout.marginHeight = 0;
+		layout.marginWidth = 0;
+		container.setLayout(layout);
+
+		newButton = createButton(container, MESSAGE_NEW_BUTTON);
+		editButton = createButton(container, MESSAGE_EDIT_BUTTON);
+		duplicateButton = createButton(container, MESSAGE_DUPLICATE_BUTTON);
+		removeButton = createButton(container, MESSAGE_REMOVE_BUTTON);
+		importButton = createButton(container, MESSAGE_IMPORT_FILES_BUTTON);
+		exportButton = createButton(container, MESSAGE_EXPORT_FILE_BUTTON);
+	}
+
+	private Button createButton(Composite parent, String text) {
+		Button button = new Button(parent, SWT.NONE);
+		button.setText(text);
+		button.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, false));
+		return button;
 	}
 
 	// TODO: call this function when "New" or "Edit" button clicked
 	private void openPresetEditingWizardFor(IPreset preset) {
-		if (DialogToolkit.openWizardWithHelp(new PresetEditingWizard(preset))) {
-			// TODO: save the modified preset to the repository
+		if (!DialogToolkit.openWizardWithHelp(new PresetEditingWizard(preset))) {
+			return;
+		}
+
+		// TODO: save the modified preset to the repository 
+		repository.add(preset);
+	}
+
+	private void bindListeners() {
+		// TODO: should the repository be Observable and refresh the view when it changes?
+		newButton.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				openPresetEditingWizardFor(new Preset());
+				tableViewer.refresh();
+			}
+		});
+
+		editButton.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				openPresetEditingWizardFor((IPreset) tableViewer.getStructuredSelection().getFirstElement());
```

Comment:
Also when I try and edit a file, instead of just modifying the current file it also creates a new one

Initial view:
![first_preset](https://user-images.githubusercontent.com/29706926/86654111-b54d1380-bfb3-11ea-953a-065e82284057.png)
Editing the file:
![second_preset](https://user-images.githubusercontent.com/29706926/86654122-b7af6d80-bfb3-11ea-9e37-3a5bfda879bf.png)

Two files shown instead of one:
![third_preset](https://user-images.githubusercontent.com/29706926/86654152-bbdb8b00-bfb3-11ea-8150-1644ee631401.png)